### PR TITLE
build: add export targets

### PR DESCRIPTION
**Pull request by compnerd**
This exports targets from SPM so that they can be used in SourceKit-LSP.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please test

File: cmake/modules/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,7 @@
+
```
Comment (aciidb0mb3r) [inline cmake/modules/CMakeLists.txt]: Please add copyright header

File: cmake/modules/SwiftPMConfig.cmake.in [hunk 2]
```diff
@@ -0,0 +1,4 @@
+
```
Comment (aciidb0mb3r) [inline cmake/modules/SwiftPMConfig.cmake.in]: Please add copyright header

File: cmake/modules/SwiftPMConfig.cmake.in [hunk 3]
```diff
@@ -0,0 +1,4 @@
...
+if(NOT TARGET Build)
```
Comment (aciidb0mb3r) [inline cmake/modules/SwiftPMConfig.cmake.in]: why this specific target?

Comment (compnerd) [reply cmake/modules/SwiftPMConfig.cmake.in]: I'm happy to change it to any other target, thats what I had in my buffer at the time :-p

**Comment by compnerd** [comment]
CC: @benlangmuir

**Comment by tomerd** [comment]
hi @compnerd is this still PR relevant?

**Comment by compnerd** [comment]
Yes, this is relevant for LSP.

**Comment by tomerd** [comment]
@swift-ci please test

**Comment by tomerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
Retested this, it works and enables SourceKit-LSP work.

**Pull request merged**